jumpstarter/resources.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import typing
+from functools import partial, wraps
+
+import anyio
+import transitions
+from anyio.abc import CapacityLimiter
+from wrapt import ObjectProxy
+
+from jumpstarter.states import ActorStartingState
+
+__all__ = (
+    "NotAResourceError",
+    "ResourceAlreadyExistsError",
+    "ResourceUnavailable",
+    "resource",
+    "ThreadedContextManager",
+)
+
+
+class NotAResourceError(Exception):
+    def __init__(self, resource_name: str, return_value):
+        super(NotAResourceError, self).__init__(
```

Comment:
is explicit call to class needed here inside super??

Reply:
No, that's how pycharm autocompletes it. I'll fix this.